narwhals/_arrow/series.py

```diff
@@ -304,7 +302,11 @@ def n_unique(self) -> int:
         return pc.count(unique_values, mode="all")  # type: ignore[no-any-return]
 
     def __native_namespace__(self) -> Any:  # pragma: no cover
-        return get_pyarrow()
+        if self._implementation is Implementation.PYARROW:
+            return self._implementation.to_native_namespace()
```

Comment:
while we're here, could we add a test which hits this? refactors with `# pragma: no cover` are always a bit scary 😄

Reply:
I removed the `# pragma: no cover` and shows to be covered (somewhere).
Would you prefer having dedicated tests?

Reply:
naah so long as it's covered it's fine 🤠